database/models/sensors.js

```diff
@@ -5,38 +5,67 @@ sql.setDialect('postgres');
 var databaseP = require('../management/databaseClientP');
 var getRandomName = require('pokemon-names').random;
 
-var sensors = require('../management/declarations.js').sensors;
-var sensorCache = 'RIEN';
+var sensorTable = require('../management/declarations.js').sensors;
+var outputTable = require('../management/declarations.js').outputs;
 
 module.exports = {
-    cache: function(){
-        return sensorCache;
-    },
     create: function (data) {
-        sensorCache = 'TEST';
 
+        var sensorData = {};
+        var types = [];
+
+        Object.keys(data).forEach(function(key){
+            if (key === 'outputs')
+                types = data[key];
+            else
+                sensorData[key] = data[key];     
+        });
+
+        // first create the sensor
         return databaseP.then(function (db) {
 
-            if (data.sim === undefined || (typeof(data.sim) === 'string' && !data.sim.length)) {
+            if (sensorData.sim === undefined || (typeof(sensorData.sim) === 'string' && !sensorData.sim.length)) {
```

Comment:
can be reduced to `if(typeof sensorData.sim !== 'string' || sensorData.sim.length === 0)`